samplers/parser.go

```diff
@@ -69,12 +70,48 @@ func ConvertMetrics(m *protocol.Message) ([]UDPMetric, error) {
 		}
 		metrics = append(metrics, metric)
 	}
+	indicators, err := metricsFromIndicatorSpan(m)
+	if err == nil {
+		metrics = append(metrics, indicators...)
+	}
 	if len(invalid) != 0 {
 		return metrics, &invalidMetrics{invalid}
 	}
 	return metrics, nil
 }
 
+const timerMetric = "ssf.indicator_duration_ms"
+
+func metricsFromIndicatorSpan(m *protocol.Message) ([]UDPMetric, error) {
+	metrics := []UDPMetric{}
+	span, err := m.TraceSpan()
+	if err != nil {
+		return metrics, err
+	}
+	if !span.Indicator {
+		// No-op if this isn't an indicator span
+		return metrics, nil
+	}
+
+	end := time.Unix(span.EndTimestamp/1e9, span.EndTimestamp%1e9)
+	start := time.Unix(span.StartTimestamp/1e9, span.StartTimestamp%1e9)
+	ssfTimer := &ssf.SSFSample{
+		Metric: ssf.SSFSample_HISTOGRAM,
+		Name:   timerMetric,
+		Value:  float32(end.Sub(start) * time.Millisecond),
+		Tags: map[string]string{
+			"name":    span.Name,
+			"service": span.Service,
```

Comment:
We'll also need the name of the span, because a single service may have more than one indicator span (although most will have one).

Reply:
that's happening one line above (103) (:

Reply:
...haha, of course. Carry on!